config.go

```diff
@@ -43,5 +48,41 @@ func (d *DestinationConfig) validateConfig() error {
 	if d.AuthMechanism == AuthMechanismBasic && (d.AuthUsername == "" || d.AuthPassword == "") {
 		return fmt.Errorf("auth.basic.username and auth.basic.password should be provided for basic authentication mechanism")
 	}
+	err := d.validateNodes()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *DestinationConfig) validateNodes() error {
+	var err error
+	for _, n := range d.Nodes {
+		err = d.validateHostPort(n)
+		if err != nil {
+			return fmt.Errorf("invalid node format %q: %w", n, err)
+		}
+	}
+	return nil
+}
+
+func (d *DestinationConfig) validateHost(host string) error {
+	hostRegexRFC952 := regexp.MustCompile(`^[a-zA-Z]([a-zA-Z0-9\-]+[\.]?)*[a-zA-Z0-9]$`)
+	hostRegexRFC1123 := regexp.MustCompile(`^([a-zA-Z0-9]{1}[a-zA-Z0-9-]{0,62}){1}(\.[a-zA-Z0-9]{1}[a-zA-Z0-9-]{0,62})*?$`)
```

Comment:
We can move these into global variables, so we don't recompile the regex every time this function is called.
Also, both regexes seem to be very similar, we should probably choose one. Seems like the second one accepts more values (iiuc it allows numbers in front of the hostname), if Cassandra is fine with that we can just use that one.

Reply:
IIUC after doing more research, then compliance with RFC952 is a necessary but not sufficient condition for compliance with RFC1123. Also, an address can comply to RFC952 but not to RFC1123.
However, RFC1123 is generally considered the de facto standard for internet host requirements these days, so I'll go with that.